packages/block-editor/src/components/list-view/block.js

```diff
@@ -220,6 +221,14 @@ function ListViewBlock( {
 		? selectedClientIds
 		: [ clientId ];
 
+	// When the block is selected, pass in the ref to the selected item, so that it can be scrolled into view.
+	// For long lists, the placeholder for the selected block is also observed, within ListViewBranch.
+	useListViewScrollIntoView( {
+		firstSelectedBlockClientId: selectedClientIds?.[ 0 ],
+		numBlocksSelected: selectedClientIds?.length,
+		selectedItemRef: isSelected ? cellRef : undefined,
```

Comment:
There's a small inconsistency in that in one place the ref is bound to a row, and in the other a cell. I think it'd be tidied up by only calling `useListViewScrollIntoView` in `ListViewBranch` (removing it here) and having `ListViewBlock` accept a ref to its row.